src/dso_api/dynamic_api/serializers.py

```diff
@@ -388,14 +398,19 @@ def get_view_name(model: Type[DynamicModel], suffix: str):
 
 
 @lru_cache()
-def serializer_factory(
+def serializer_factory(  # noqa: C901
     model: Type[DynamicModel], depth: int, flat=None, links_serializer=False
 ) -> Type[DynamicSerializer]:
     """Generate the DRF serializer class for a specific dataset model.
     It can generate the serializer for both the body as well as for the _links field,
     depending on links_serializer being False or True."""
+    if isinstance(model, str):
+        raise ImproperlyConfigured(f"Model {model} could not be resolved.")
```

Comment:
Nice addition, though this check should be probably done before `serializer_factory` is called, as having model name is not enough to find root cause of a problem.

Reply:
apparently that was already in the code, but below the `fields = [..]` logic..

Reply:
Yeah, it's an old one and definitely outside of cleanup PR ;)